crates/oxide/src/extractor/pre_processors/ruby.rs

```diff
@@ -77,6 +77,72 @@ impl PreProcessor for Ruby {
 
         // Ruby extraction
         while cursor.pos < len {
+            match cursor.curr {
+                b'"' => {
+                    cursor.advance();
+
+                    while cursor.pos < len {
+                        match cursor.curr {
+                            // Escaped character, skip ahead to the next character
+                            b'\\' => cursor.advance_twice(),
+
+                            // End of the string
+                            b'"' => break,
+
+                            // Everything else is valid
+                            _ => cursor.advance(),
+                        };
+                    }
+
+                    cursor.advance();
+                    continue;
+                }
+
+                b'\'' => {
+                    cursor.advance();
+
+                    while cursor.pos < len {
+                        match cursor.curr {
+                            // Escaped character, skip ahead to the next character
+                            b'\\' => cursor.advance_twice(),
+
+                            // End of the string
+                            b'\'' => break,
+
+                            // Everything else is valid
+                            _ => cursor.advance(),
+                        };
+                    }
+
+                    cursor.advance();
+                    continue;
+                }
```

Comment:
Can we add some tests where we handle quotes? We have to be careful with these because it could be that you start a sentence with a `'` in it and never close it.

```haml
%p a quote here can't exist, or can it 
-#
  This won't be displayed (there is a quote here as well)
    Nor will this
                   Nor will this.
%p a quote here can't exist, or can it
```

Reply:
I've added tests and updated the implementation a bit